src/include/stir/IO/InputFileFormatSAFIR.h

```diff
@@ -0,0 +1,147 @@
+/* InputFileFormatSAFIR.h
+
+ Class defining input file format for coincidence listmode data for SAFIR. 
+ Jannis Fischer
+ jannis.fischer@cern.ch
+
+	Copyright 2015 ETH Zurich, Institute of Particle Physics
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+ */
+
+/*!
+
+  \file
+  \ingroup listmode
+  \brief Declaration of class stir::SAFIRCListmodeInputFileFormat
+
+  \author Jannis Fischer
+*/
+
+#ifndef __local_SAFIR_InputFileFormatSAFIR_H__
+#define __local_SAFIR_InputFileFormatSAFIR_H__
+
+#include <cstring>
+#include <string>
+#include <iostream>
+
+#include "stir/IO/InputFileFormat.h"
+#include "stir/utilities.h"
+#include "stir/ParsingObject.h"
+
+#include "stir/listmode/CListRecordSAFIR.h"
+#include "stir/listmode/CListModeDataSAFIR.h"
+
+START_NAMESPACE_STIR
+
+//! Class for reading SAFIR coincidence listmode data.
+class SAFIRCListmodeInputFileFormat : public InputFileFormat<CListModeData>, public ParsingObject
+{
+public:
+	SAFIRCListmodeInputFileFormat() : did_parsing(false) {}
+	virtual const std::string get_name() const
+	{
+		return "SAFIR Coincidence Listmode File Format";
+	}
+	
+	//! Checks in binary data file for correct signature.
+	virtual bool can_read(const FileSignature& signature, std::istream& input ) const
+	{
+		return this->actual_can_read(signature, input);
+	}
+
+	//! Checks in binary data file for correct signature (can be either "SAFIR CListModeData" or "MUPET CListModeData").
+	virtual bool can_read( const FileSignature& signature, const std::string& filename) const
+	{
+		actual_do_parsing(filename);
```

Comment:
This is not easily possible, I fear. I have to parse the input parameter file do actually know the filename of the binary file, where I can check the header. This was the solution we came up with because I actually need 3 files (crystal map, binary data, projdata template). Or am I missing something?